lms/djangoapps/certificates/apis/v0/permissions.py

```diff
@@ -0,0 +1,29 @@
+"""
+This module provides a custom DRF Permission class for supporting the course certificates
+to Admin users and users whom they belongs to.
+"""
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from rest_framework.permissions import BasePermission
+
+from openedx.core.djangoapps.user_api.accounts.serializers import get_profile_visibility
+
+
+class IsAdminOrSelfUser(BasePermission):
+    """
+    Method that will ensure whether the requesting user is staff or
+    the user whom the certificate belongs to
+    """
+    def has_permission(self, request, view):
+        user = request.user
+        requested_profile_username = view.kwargs.get('username')
+
+        if user.is_staff or user.username == requested_profile_username:
```

Comment:
I think we don't need to check for staff here, we can use `IsAdminUser` for that purpose.

Reply:
@nasthagiri: Would it be reasonable to use [JwtHasUserFilterForRequestedUser](https://github.com/edx/edx-drf-extensions/blob/master/edx_rest_framework_extensions/permissions.py#L115) for this?

Reply:
@robrap Makes sense if we guarantee the view is always authenticated with a JWT right?  Also, I don't believe that permission allows for staff users.  Are you imagining it would be combined with a Staff-checking permission?

Reply:
@nasthagiri: 
1. May need some thought.  It would be nice to have a permission that ensures the JWT has the correct filter when using a JWT, and does the simpler check when not JwtAuthenticated.
2. Yes, combining with `IsAdminUser` like @waheedahmed had mentioned.